ПР №2 - Домашнее задание по Спринту 9

asdaaarr opened the pull request:
kucherencko.denis@yandex.ru

asdaaarr committed:
Коммит №2 - Домашнее задание по Спринту 9

Charmandik commented on app/src/main/java/com/practicum/playlistmaker/SearchActivity.kt:
```diff
@@ -1,20 +1,89 @@
-            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
-            insets
+
+        val backButton = findViewById<ImageView>(R.id.btn_back)
```
Попробуй применить подход View Binding. ViewBinding позволяет проще писать код для взаимодействия с версткой позволяет избежать этого процесса с инициализацией каждой view по отдельности. Более подробно можно почитать в статье https://habr.com/ru/articles/467295/

asdaaarr replied:
Не знаю где тебе написать
У меня есть замечание по заданию
⚠️ Критические замечания:
❌ Цвета, отступы и шрифты соответствуют дизайну.
Цвет подсказки на экране Поиск не сответствует дизайну

Я посмотрел и не понял, в чем проблема
Для подсказки у меня задано 2 цвета (светлая/темная тема) в colors.xml
В коде это для светлой:
<color name="text_color_search_hint">#AEAFB4</color>
Для темной
<color name="text_color_search_hint">#1A1B22</color>

Эти цвета я и вижу в макетах в фигме, может я что-то не понимаю?

Charmandik commented on app/src/main/java/com/practicum/playlistmaker/SearchActivity.kt:
```diff
@@ -1,20 +1,89 @@
+
+    private fun clearButtonVisibility(s: CharSequence?): Int {
+        return if (s.isNullOrEmpty()) {
+                View.GONE
```
Для управления видимостью можно использовать свойство isVisible, устанавливая значение false для View.GONE и true для View.VISIBLE
Например, если нам нужно установить View.VISIBLE:
myViewName.isVisible = true
Или если нужно установить View.GONE, то : 
myViewName.isVisible = false

Возможно, тебе больше понравится такой подход

Charmandik commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -1,10 +1,88 @@
+        android:orientation="horizontal"
+        android:gravity="center_vertical"
+        android:paddingTop="14dp"
+        android:paddingBottom="@dimen/indent_16"
```
В дизайне могут быть нечетные размеры, или не симметричные. Но лучше использовать в коде размеры, кратные 4 или 8. Документация - https://material.io/design/layout/spacing-methods.html#baseline-grid

Charmandik commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -1,10 +1,88 @@
+    <FrameLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:paddingTop="@dimen/indent_8"
```
Если есть одинаковые android:paddingTop и android:paddingBottom, можно использовать paddingVertical

Charmandik commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -1,10 +1,88 @@
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:background="@color/background_color_search"
+    android:orientation="vertical"
+    android:paddingStart="@dimen/indent_16"
```
Если есть одинаковые paddingStart и paddingEnd, можно использовать paddingHorizontal

asdaaarr committed:
Commit №3 - Домашнее задание по Спринту 9
Исправил критические замечания + учел все рекомендации

У меня есть замечание по заданию
⚠️ Критические замечания:
❌ Цвета, отступы и шрифты соответствуют дизайну.
Цвет подсказки на экране Поиск не сответствует дизайну

Я посмотрел и не понял, в чем проблема
Для подсказки у меня задано 2 цвета (светлая/темная тема) в colors.xml
В коде это для светлой:
#AEAFB4
Для темной
#1A1B22

Эти цвета я и вижу в макетах в фигме, может что-то упускаю

asdaaarr committed:
Merge branch 'main' into dev2
